odoo/addons/base/models/ir_qweb_fields.py

```diff
@@ -398,7 +398,11 @@ def value_to_html(self, value, options):
         except: # image.verify() throws "suitable exceptions", I have no idea what they are
             raise ValueError("Invalid image content")
 
-        return Markup('<img src="data:%s;base64,%s">' % (Image.MIME[image.format], value.decode('ascii')))
+        return "data:%s;base64,%s" % (Image.MIME[image.format], value.decode('ascii'))
+
+    @api.model
+    def value_to_html(self, value, options):
+        return Markup('<img src="%s">' % self._get_src_data_b64(value, options))
```

Comment:
```suggestion
        return Markup('<img src="%s">') % self._get_src_data_b64(value, options)
```
doesn't work?

Reply:
@mart-e 
yes it does, I made the change but it doesn't solve the security warning.
Also, more of question here, but I have the intuition that formatting string after having marked up the html string would allow to inject unsafe stuff in it without needing to re-markup the result afterwards, does that make sense ?
Intuitively, I wanted the result to be marked up, because I know the result is safe since I control which are the values I inject in it, and where.

Reply:
@kebeclibre on the contrary if you want to prevent injection, you have to pass the parameters outside of the markup call
```py
>>> Markup('<a title="%s"/>') % 'haha "/><script>alert(42)</script><a '
Markup('<a title="haha &#34;/&gt;&lt;script&gt;alert(42)&lt;/script&gt;&lt;a "/>')
>>> Markup('<a title="%s"/>' % 'haha "/><script>alert(42)</script><a ')
Markup('<a title="haha "/><script>alert(42)</script><a "/>')
```

Reply:
Sweet, thanks. I guess the explanation at https://github.com/odoo/enterprise/wiki/security#PY014 needs some rewording, I don't find it very clear on what we should do